pkg/executor/cluster.go

```diff
@@ -140,44 +141,44 @@ func EnsureFinalizer(
 	if found {
 		return nil
 	}
-	cr.Finalizers = append(cr.Finalizers, finalizerID)
 
-	var finalizersPatch []finalizersPatchSpec
-	if len(cr.Finalizers) == 1 {
-		finalizersPatch = []finalizersPatchSpec{
-			{
-				Op:    "add",
-				Path:  "/metadata/finalizers",
-				Value: cr.Finalizers,
-			},
-		}
+	cr.Finalizers = append(cr.Finalizers, finalizerID)
 
-	} else {
-		finalizersPatch = []finalizersPatchSpec{
-			{
-				Op:    "replace",
-				Path:  "/metadata/finalizers",
-				Value: cr.Finalizers,
-			},
-		}
-	}
-	finalizersPatchBytes, patchErr := json.Marshal(finalizersPatch)
-	if patchErr == nil {
-		patchErr = sdk.Patch(cr, types.JSONPatchType, finalizersPatchBytes)
+	// TODO: Can we just use the cr that was passed in?
+	prevCr := &kdv1.KubeDirectorCluster{}
```

Comment:
More comments about PATCH stuff above.

Reply:
See reply to lengthy PATCH comment above.